rosbag2_transport/include/rosbag2_transport/recorder.hpp

```diff
@@ -39,25 +40,27 @@ class Writer;
 
 namespace rosbag2_transport
 {
-namespace impl
-{
 
 class Recorder : public rclcpp::Node
 {
 public:
+  ROSBAG2_TRANSPORT_PUBLIC
   explicit Recorder(
     const std::string & node_name = "rosbag2_recorder",
     const rclcpp::NodeOptions & node_options = rclcpp::NodeOptions());
 
+  ROSBAG2_TRANSPORT_PUBLIC
   Recorder(
     std::shared_ptr<rosbag2_cpp::Writer> writer,
     const rosbag2_storage::StorageOptions & storage_options,
     const rosbag2_transport::RecordOptions & record_options,
     const std::string & node_name = "rosbag2_recorder",
     const rclcpp::NodeOptions & node_options = rclcpp::NodeOptions());
 
+  ROSBAG2_TRANSPORT_PUBLIC
   virtual ~Recorder();
 
+  ROSBAG2_TRANSPORT_PUBLIC
   void record();
```

Comment:
Now that we are making these public I think we need a serious docstring pass. Not blocking this, but we should make an effort to document every public header in the "cleanup sprint"